packages/block-library/src/query-loop/index.php

```diff
@@ -26,8 +26,18 @@ function render_block_core_query_loop( $attributes, $content, $block ) {
 		'post__not_in' => array(),
 	);
 
+	$use_global_query = ( isset( $block->context['query']['customQuery'] ) && false === $block->context['query']['customQuery'] );
+
+	// Override the custom query with the global query if needed.
+	if ( $use_global_query ) {
+		global $wp_query;
+		if ( $wp_query && isset( $wp_query->query_vars ) && is_array( $wp_query->query_vars ) ) {
+			$query = $wp_query->query_vars;
+		}
+	}
+
 	if ( isset( $block->context['query'] ) ) {
-		if ( isset( $block->context['query']['postType'] ) ) {
+		if ( ! $use_global_query && isset( $block->context['query']['postType'] ) ) {
```

Comment:
You should wrap all the properties of `query context` that are affected by `$use_global_query` like:
```
if ( ! $use_global_query ) {
  if ( isset( $block->context['query']['categoryIds'] ) ) { ... }
  .....
}
```

Reply:
Done